crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -0,0 +1,974 @@
+            .open(directory.join(Self::METADATA_FILE))?;
+
+        let (metadata_header, mut metadata_header_mmap) = if metadata_file.seek(SeekFrom::End(0))?
+            == 0
+        {
+            let metadata_header = PlotMetadataHeader {
+                version: 0,
+                sector_count: 0,
+            };
+
+            metadata_file.preallocate(
+                RESERVED_PLOT_METADATA
+                    + SectorMetadata::encoded_size() as u64 * target_sector_count,
+            )?;
+            metadata_file.write_all_at(metadata_header.encode().as_slice(), 0)?;
+
+            let metadata_header_mmap = unsafe {
+                MmapOptions::new()
+                    .len(PlotMetadataHeader::encoded_size())
+                    .map_mut(&metadata_file)?
+            };
+
+            (metadata_header, metadata_header_mmap)
+        } else {
+            let metadata_header_mmap = unsafe {
+                MmapOptions::new()
+                    .len(PlotMetadataHeader::encoded_size())
+                    .map_mut(&metadata_file)?
+            };
+
+            let metadata_header = PlotMetadataHeader::decode(&mut metadata_header_mmap.as_ref())
+                .map_err(SingleDiskPlotError::FailedToDecodeMetadataHeader)?;
+
+            if metadata_header.version != 0 {
+                return Err(SingleDiskPlotError::UnexpectedMetadataVersion(
+                    metadata_header.version,
+                ));
+            }
+
+            (metadata_header, metadata_header_mmap)
+        };
+
+        metadata_file.advise_random_access()?;
+
+        let metadata_header = Arc::new(Mutex::new(metadata_header));
+
+        let mut metadata_mmap_mut = unsafe {
+            MmapOptions::new()
+                .offset(RESERVED_PLOT_METADATA)
+                .len(SectorMetadata::encoded_size() * target_sector_count as usize)
+                .map_mut(&metadata_file)?
+        };
+
+        let plot_file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .open(directory.join(Self::PLOT_FILE))?;
+
+        plot_file.preallocate(plot_sector_size * target_sector_count)?;
+        plot_file.advise_random_access()?;
+
+        let mut plot_mmap_mut = unsafe { MmapMut::map_mut(&plot_file)? };
+
+        // TODO: Use this or remove
+        let _codec = SubspaceCodec::new_with_gpu(public_key.as_ref());
+
+        let (error_sender, error_receiver) = oneshot::channel();
+        let error_sender = Arc::new(Mutex::new(Some(error_sender)));
+
+        let tasks = FuturesUnordered::<BackgroundTask>::new();
+
+        tasks.push(Box::pin(async move {
+            if let Ok(error) = error_receiver.await {
+                return Err(error);
+            }
+
+            Ok(())
+        }));
+
+        let shutting_down = Arc::new(AtomicBool::new(false));
+
+        let plotting_join_handle = thread::Builder::new()
+            .name(format!("p-{single_disk_plot_id}"))
```

Comment:
More readable to me this way, and the below `farming-{single_disk_plot_id}`

```suggestion
            .name(format!("plotting-{single_disk_plot_id}"))
```

Reply:
While certainly more readable, OS has a limit for the length of the thread/app name, I decided to use `p-` prefix instead of `plotting-` so we can see more of `single_disk_plot_id` during debugging.